### DISCUSSION TITLE
How do you manage the Client Scopes for a realm with the Operator?

### DISCUSSION THREAD

**Discussion by mmosttler** [keycloak/keycloak#8995] [Q&A] [answered]
With the Operator CR's I am able to manage clients and users, but I haven't seen how to manage the Client Scopes.  Is there a way with CR's to manage (create, change, delete) the Client Scopes? 
Is the rest api the only way to manage them outside of the UI, nothing declarative?

**Comment by vmuzikar** [comment, marked as answer]
Client Scopes can be only created [through the Realm CR](https://github.com/keycloak/keycloak-operator/blob/77d126113104d45e29057f5c9e30c54f1a482763/deploy/crds/keycloak.org_keycloakrealms_crd.yaml#L204).

**Comment by mmosttler** [comment]
The Realm CR is only able to be used for creating the realm.  That means you must know all about all your scope configs at the time of initialization.  Is there a reason that a client scope cr wasn't built?

**Reply by vmuzikar** [reply]
Unfortunately, we cannot provide a CRD for every resource in Keycloak. Due to how CRDs are implemented in the current operator (manually mapping of Keycloak representations to CRDs, REST API calls, possible conflicts with changes at the Keycloak side), this would be unmaintainable. However, we're currently developing a new operator as part of [Keycloak.X](https://www.keycloak.org/2021/10/keycloak-x-update.html) where we'll be reconsidering our CRDs approach. More on details on the new operator soon. :) Thanks for your understanding.
